api/src/transactions.rs

```diff
@@ -404,17 +404,14 @@ impl Transactions {
             return Response::new(self.ledger_info, &Vec::<Transaction>::new());
         }
 
-        let first_version = data[0].version;
-        let mut timestamp = self.context.get_block_timestamp(first_version)?;
         let resolver = self.context.move_resolver()?;
         let converter = resolver.as_converter();
         let txns: Vec<Transaction> = data
             .into_iter()
             .map(|t| {
+                let version = t.version;
+                let timestamp = self.context.get_block_timestamp(version)?;
```

Comment:
Are we calling this too much? 

Use to be an explicit optimization for calling it less.. what's wrong with it?

Reply:
the transactions are not necessarily chronological. if we try to render transactions by accounts, the timestamp never gets updated